### PR TITLE
Converted StorageBackend::put to async/.await

### PR DESCRIPTION
**Pull request by hannesdejager**
Another commit towards #120 and #137

### CODE REVIEW

File: src/storage/cloud_storage/mod.rs [hunk 1]
```diff
@@ -290,40 +290,39 @@ impl<U: Sync + Send> StorageBackend<U> for CloudStorage {
...
         bytes: B,
         path: P,
         _start_pos: u64,
-    ) -> Box<dyn Future<Item = u64, Error = Error> + Send> {
-        let uri = match self.uris.put(path) {
-            Ok(uri) => uri,
-            Err(err) => return Box::new(future::err(err)),
-        };
+    ) -> super::Result<u64> {
+        use futures03::compat::*;
```
Comment (Nonius) [inline src/storage/cloud_storage/mod.rs:300]: this one is already at the top of the file, but we need a cleanup anyway after everything i s converted, so it doesn't really matter

File: src/storage/filesystem.rs [hunk 2]
```diff
@@ -182,7 +182,9 @@ impl<U: Send + Sync> StorageBackend<U> for Filesystem {
         use tokio02::fs::OpenOptions;
         use tokio02util::compat::FuturesAsyncReadCompatExt;
 
-        let fut01 = async move {
+        // TODO: Remove async block
+        // TODO: Convert to new style streams
+        async move {
```
Comment (Nonius) [inline src/storage/filesystem.rs:187]: Is this async move still needed? This is already inside an async function.

File: src/storage/cloud_storage/mod.rs (hunk 1 above)
Comment (hannesdejager) [reply src/storage/cloud_storage/mod.rs]: Removed

File: src/storage/filesystem.rs (hunk 2 above)
Comment (hannesdejager) [reply src/storage/filesystem.rs:187]: I'll do a seperate commit

**Pull request merged**